internal/objects/deployment/deployment.go

```diff
@@ -132,6 +134,10 @@ func DesiredDeployment(contour *operatorv1alpha1.Contour, image string) *appsv1.
 			args = append(args, fmt.Sprintf("--envoy-service-https-port=%d", port.PortNumber))
 		}
 	}
+	// Pass the IngressClass name if it's not the default.
+	if contour.Spec.IngressClassName != ContourIngressClassName {
```

Comment:
Ah, I said earlier that it couldn't have a default, if we do it like this, we can have a `contour` default, I suppose.

Reply:
Yeah, I prefer being explicit with fields over using pointers.

Reply:
If someone explicitly sets the field in YAML to `""` it won't be overriden by the default in the kubebuilder validation setup right? Should we also check the value isn't `""` here?

Reply:
> If someone explicitly sets the field in YAML to "" it won't be overriden by the default in the kubebuilder validation setup right? Should we also check the value isn't "" here?

@sunjayBhatia a Contour with `ingressClassName: ""` would fail api validation due to `+kubebuilder:validation:MinLength=1`.

Example:
```
$ cat <<EOF | kubectl apply -f -
> apiVersion: operator.projectcontour.io/v1alpha1
> kind: Contour
> metadata:
>   name: contour-sample
> spec:
>   ingressClassName: ""
>   networkPublishing:
>     envoy:
>       type: NodePortService
> EOF
The Contour "contour-sample" is invalid: spec.ingressClassName: Invalid value: "": spec.ingressClassName in body should be at least 1 chars long
```

Reply:
Ha! good call